docs/cow-protocol/tutorials/arbitrate/solver/driver.md

```diff
@@ -128,6 +128,45 @@ sequenceDiagram
     deactivate Driver
 ```
 
+#### Flash Loans Encoding
+
+If a solver decides to encode the transaction without the help of the reference driver, the solver must call the `IFlashLoanRouter` contract's [flashLoanAndSettle](../../../reference/contracts/periphery/flash-loans.md#flashloanandsettle) function instead of the settlement contract's [settle](../../../reference/contracts/core/settlement.md#settle) function. The solver must provide all necessary flash loan inputs for the settlement, as well as the settle calldata, which will be executed within the same context by the `IFlashLoanRouter` contract. The `IFlashLoanRouter` contract will then request the specified flash loans and, once received, execute the settlement as instructed.
+
+The entry point to the router contract ([IFlashLoanRouter](../../../reference/contracts/periphery/flash-loans.md#iflashloanrouter-contract)) is the function `flashLoanAndSettle`.
+It takes a list of loans with the following entries for each loan:
+
+- The loaned amount and ERC-20 token.
+- The flash-loan lender (e.g., Balancer, Aave, Maker, ...).
+- The _borrower_, which is an adapter contract that makes the specific lender implementation compatible with the router.
+
+It also takes the exact call data for a call to `settle`.
+The flash-loan router is a solver for CoW Protocol and calls `settle` directly once the flash loans have been obtained.
+Only CoW-Protocol solvers can call this function.
+
+Tokens and lenders are external contracts, while the router and borrowers have a dedicated contract implemented by CoW Protocol.
```

Comment:
nit: hopefully not all borrower contracts will be implemented by us. For example external integrations should write their own adapter contracts. In those cases we only need to vet the contracts somehow.